File: 1319-unique-number-of-occurrences/1319-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int n = arr.length;
        for(int i = 0;i<n;i++){
            if(map.containsKey(arr[i])){
                int incr = map.getOrDefault(arr[i],0) + 1;
                map.put(arr[i],incr);
            }
            else{
                map.put(arr[i],1);
            }
        }

        HashSet<Integer> set = new HashSet<>();
        boolean ans = true;
        for (int val : map.values()) {
            if (set.contains(val)) {
                ans = false;
                break;
            }
            set.add(val);
        }

        return ans;
    }
}

File: 1319-unique-number-of-occurrences/1319-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> map;
        int n = arr.size();

        for (int i = 0; i < n; i++) {
            if (map.count(arr[i]) > 0) {
                int incr = map[arr[i]] + 1;
                map[arr[i]] = incr;
            }
            else {
                map[arr[i]] = 1;
            }
        }

        unordered_set<int> set;
        bool ans = true;

        for (const auto& val : map) {
            if (set.count(val.second) > 0) {
                ans = false;
                break;
            }
            set.insert(val.second);
        }

        return ans;
    }
};